packages/reactotron-react-native/src/plugins/networking.js

```diff
@@ -74,40 +74,51 @@ export default (pluginConfig = {}) => reactotron => {
       headers: xhr._headers || null
     }
 
-    // assemble the response object
-    let body = null
-
     // what type of content is this?
     const contentType =
       (xhr.responseHeaders && xhr.responseHeaders['content-type']) ||
       (xhr.responseHeaders && xhr.responseHeaders['Content-Type']) ||
       ''
 
+    const sendResponse = responseBodyText => {
+      let body = `~~~ skipped ~~~`
+      if (responseBodyText && responseBodyText.length) {
+        try {
+          // all i am saying, is give JSON a chance...
+          body = JSON.parse(responseBodyText)
+        } catch (boom) {
+          body = response
+        }
+      }
+      const tronResponse = {
+        body,
+        status,
+        headers: xhr.responseHeaders || null
+      }
+
+      // send this off to Reactotron
+      reactotron.apiResponse(tronRequest, tronResponse, stopTimer())
+    }
+
     // can we use the real response?
     const useRealResponse =
       (typeof response === 'string' || typeof response === 'object') &&
       !ignoreContentTypes.test(contentType || '')
 
     // prepare the right body to send
     if (useRealResponse) {
-      try {
-        // all i am saying, is give JSON a chance...
-        body = JSON.parse(response)
-      } catch (boom) {
-        body = response
+      if (type === 'blob' && typeof FileReader !== 'undefined') {
+        // Disable reason: FileReader should be in global scope since RN 0.54
+        // eslint-disable-next-line no-undef
+        const bReader = new FileReader()
+        bReader.addEventListener('loadend', () => sendResponse(bReader.result))
```

Comment:
Do we have to cleanup after ourselves for this listener or does it auto cleanup listeners?

Reply:
It appears `FileReader` inherits from `EventTarget` which I believe is this package:
https://www.npmjs.com/package/event-target-shim

If that is the case it should also provide a `removeEventListener` method. If that is case this might be as easy calling that with the correct arguments whenever it makes the most sense to.